clippy_lints/src/ptr_eq.rs

```diff
@@ -0,0 +1,96 @@
+use crate::utils;
+use if_chain::if_chain;
+use rustc::hir::{BinOpKind, Expr, ExprKind};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::{declare_lint_pass, declare_tool_lint};
+use rustc_errors::Applicability;
+
+declare_clippy_lint! {
+    /// **What it does:** Use `std::ptr::eq` when applicable
+    ///
+    /// **Why is this bad?** This can be used to compare `&T` references
```

Comment:
What's `This`, the first word, meant to reference? `ptr::eq`? If yes, I think it would read better as `ptr::eq can be used to ...`